android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt

```diff
@@ -309,10 +309,34 @@ class ChargebeeFlutterSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAwar
 fun CBProduct.toMap(): Map<String, Any> {
     return mapOf(
         "productId" to productId,
-        "productPrice" to productPrice,
+        "productPrice" to convertPriceAmountInMicros(),
+        "productPriceString" to productPrice,
         "productTitle" to productTitle,
-        "currencyCode" to skuDetails.priceCurrencyCode
+        "currencyCode" to skuDetails.priceCurrencyCode,
+        "subscriptionPeriod" to subscriptionPeriod()
     )
 }
 
+fun CBProduct.convertPriceAmountInMicros(): Double {
+    return skuDetails.priceAmountMicros/1_000_000.0
+}
+
+fun CBProduct.subscriptionPeriod(): Map<String, Any> {
+    val subscriptionPeriod = skuDetails.subscriptionPeriod
+    val numberOfUnits = subscriptionPeriod[subscriptionPeriod.length / 2].digitToInt()
+    return mapOf(
+        "periodUnit" to periodUnit(),
+        "numberOfUnits" to numberOfUnits
+    )
+}
+
+fun CBProduct.periodUnit(): String {
+    return when (skuDetails.subscriptionPeriod.last().toString()) {
+        "Y" -> "year"
+        "M" -> "month"
+        "W" -> "week"
+        "D" -> "day"
+        else -> "unknown"
```

Comment:
For iOS, we are returning an empty `""` for unknown cases. Can we please have it consistent here as well?